Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -624,12 +630,45 @@ void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* m, CGPoint point, bool eoFill) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (m) {
+        point = CGPointApplyAffineTransform(point, *m);
+    }
+    // check if the point is outside this box already, if it is, return false
+    CGRect boundingBox = CGPathGetBoundingBox(path);
+
+    if ((boundingBox.origin.x > point.x) || (boundingBox.origin.y > point.y) ||
+        ((boundingBox.origin.x + boundingBox.size.width) < point.x) || ((boundingBox.origin.y + boundingBox.size.height) < point.y)) {
+        return false;
+    }
+
+    CGContextRef context =
+        CGBitmapContextCreate(0, boundingBox.origin.x + boundingBox.size.width, boundingBox.origin.y + boundingBox.size.height, 1, 1, 0, 0);
+
+    CGContextSetGrayFillColor(context, 1.0, 1.0);
+    CGContextAddPath(context, path);
+
+    if (eoFill) {
+        CGContextEOFillPath(context);
+    } else {
+        CGContextFillPath(context);
+    }
+
+    CGImageRef image = context->Backing()->DestImage();
+
+    float r = 0.0f;
+    float g = 0.0f;
+    float b = 0.0f;
+    float a = 0.0f;
+    image->Backing()->GetPixel(point.x, point.y, r, g, b, a);
```

Comment:
Oh, I agree absolutely that we should stick to Apple's API on the surface. I don't propose changing the interface to CGPathContainsPoint. What I was proposing was that, after we create the CGContext here, we delegate the rest of the logic to the CGContext object (via my hypothetical _CGContextIsPointInPath function). Since our CGContext objects are implemented on top of Cairo, the CGContext object could then turn around and call cairo_in_fill.

I doubt it would make a huge difference one way or the other, but if there's already a dedicated function for this, I figure it's probably the best route.

---
In reply to: [68843448](https://github.com/Microsoft/WinObjC/pull/587#discussion_r68843448) [](ancestors = 68843448)